### PR TITLE
Import Wiki Content into Project

### LINKED ISSUE #140: Migrate from wiki to markdown documentation

### PR DESCRIPTION
**Pull request by sputnik27**
Fixes #140

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,8 +9,8 @@ data with [OSIAM](https://github.com/osiam/osiam). It is also used by the
 [connector4java](https://github.com/osiam/connector4java) that gives you an easy
 way to interact with OSIAM, if you are using Java.
 
-For more information and documentation visit the
-[repo's Wiki](https://github.com/osiam/scim-schema/wiki).
+For more information and documentation look in the
```
Comment (wallner) [inline README.md]: For more information and documentation please take a look in the.

**Comment by dacrome** [comment]
I did not find a best practice about markdown filenames, but maybe we shouldn't use CamelCase?

**Comment by sputnik27** [comment]
> I did not find a best practice about markdown filenames, but maybe we shouldn't use CamelCase?

I must admit that I just took the file names as the gh wiki "export" spit it out, without thinking about it. The names in `api` are camel case (upper style). The ones in `schema` use dashes. We really should unify this.

Which would you prefer?
* `camelCase`
* `CamelCaseUpper`
* `with-dashes`
* `snake_case`

**Comment by dacrome** [comment]
I prefer `with-dashes`. What do you think @tkrille, @wallner?

**Comment by tkrille** [comment]
Use the naming style that's already used in all other docs to be consistent.

**Comment by tkrille** [comment]
@dacrome I've assigned you, because it appears to me, that you're reviewing this.

**Comment by sputnik27** [comment]
> Use the naming style that's already used in all other docs to be consistent.

If only we had a consistent style..

`snake_case`:
https://github.com/osiam/auth-server/tree/master/docs
https://github.com/osiam/resource-server/tree/master/docs

`with-dashes`:
https://github.com/osiam/addon-self-administration/tree/master/docs

**Comment by wallner** [comment]
I agree with @dacrome Let's go for `with-dashes`.

**Comment by tkrille** [comment]
Ok, as we don't have an existing, consistent style, I'm also in favor for `hyphenated-snake-case`.

**Comment by sputnik27** [comment]
Anyone cares to merge this?

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+####[Overview](schema/scim-schema-overview.md)
+####[Integrate the SCIM Schema in your Project](schema/scim-schema-integration.md)
+
+### API Documentation
+
+####[User](api/user.md)
+####[Group](api/group.md)
+####[Update User](api/update-user.md)
+####[Update Group](api/update-group.md)
+####[Scim Helper](api/scim-helper.md)
```
Comment (dacrome) [inline docs/README.md]: Please list the items on this level with [bullet points](http://daringfireball.net/projects/markdown/syntax#list)

File: docs/api/group.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
+A org.osiam.resources.scim **Group** Object holds all relevant data from a Group.
+
+At this wiki we only document the getter and setter of the single attributes.
+For a explanation about the "meaning" of the single fields please have a look at the [documentation](http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-8) of [System for Cross-domain Identity Management](http://tools.ietf.org/html/draft-ietf-scim-core-schema-02)
```
Comment (dacrome) [inline docs/api/group.md]: Maybe link to the [newest draft](http://tools.ietf.org/html/draft-ietf-scim-core-schema-22)?

File: docs/api/user.md [hunk 4]
```diff
@@ -0,0 +1,590 @@
+A org.osiam.resources.scim **User** Object holds all relevant data from a User.
+
+At this wiki we only document the getter and setter of the single attributes.
+For a explanation about the "meaning" of the single fields please have a look at the [documentation](http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-6) of [System for Cross-domain Identity Management](http://tools.ietf.org/html/draft-ietf-scim-core-schema-02)
```
Comment (dacrome) [inline docs/api/user.md]: Maybe link to the [newest draft](http://tools.ietf.org/html/draft-ietf-scim-core-schema-22)?

File: docs/schema/scim-schema-overview.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+The OSIAM scim schema provide you a [scim](http://tools.ietf.org/html/draft-ietf-scim-core-schema-02) conform User and Group Object.
```
Comment (dacrome) [inline docs/schema/scim-schema-overview.md]: Maybe link to the [newest draft](http://tools.ietf.org/html/draft-ietf-scim-core-schema-22)?

File: docs/schema/scim-schema-overview.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+
+These Objects are used by the [OSIAM Server](https://github.com/osiam/server) to get and to provide any User or Group Data.
```
Comment (dacrome) [inline docs/schema/scim-schema-overview.md]: ...used by the [OSIAM resource-server](https://github.com/osiam/resource-server) to get...

File: docs/schema/scim-schema-integration.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+To be able to use OSIAM in your project you need to do the following things.
+
+Install the OSIAM server. The needed Information can be found [here](https://github.com/osiam/server/wiki/detailed_reference_installation).
```
Comment (dacrome) [inline docs/schema/scim-schema-integration.md]: ...can be found [here](https://github.com/osiam/osiam/blob/master/docs/detailed-reference-installation.md).

File: docs/schema/scim-schema-integration.md [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+
+Include the OSIAM scim schema in your pom.xml. The dependencie can be found [here](http://search.maven.org/#search|ga|1|osiam).
```
Comment (dacrome) [inline docs/schema/scim-schema-integration.md]: dependency

**Comment by dacrome** [comment]
You forgot to migrate the [`Migration`](https://github.com/osiam/scim-schema/wiki/Migration) and [`LTS`](https://github.com/osiam/scim-schema/wiki/LTS) description

**Comment by sputnik27** [comment]
> You forgot to migrate the Migration and LTS description

They look extremely outdated. Are they still relevant?

**Comment by dacrome** [comment]
> They look extremely outdated. Are they still relevant?

About LTS we should discuss in general, but the migration guide should be migrated. I'm not sure if since version 1.0 is any migration needed, but maybe in future releases. Please fix also the links in the [CHANGELOG.md](CHANGELOG.md).

**Comment by dacrome** [comment]
Thanks for your update. I guess we don't need to migrate the LTS wiki page, because a LTS description already existing in the [osiam umbrella repo](https://github.com/osiam/osiam/blob/master/docs/LTS.md). What do you think @tkrille @wallner?

**Comment by wallner** [comment]
We don't need to migrate that page it's in the umbrella project. Although I think we should discuss the idea of the LTS release or at least update the document in the umbrella repo.

**Comment by dacrome** [comment]
Yeah! Thanks!

**Pull request merged**